dbt_platform_helper/utils/validation.py

```diff
@@ -553,8 +559,91 @@ def validate_platform_config(config, disable_aws_validation=False):
     validate_database_copy_section(enriched_config)
+
     if not disable_aws_validation:
         _validate_s3_bucket_uniqueness(enriched_config)
+        _validate_redis_versions(config)
+        _validate_opensearch_versions(config)
+
+
+def _validate_redis_versions(config):
+
+    supported_redis_versions = []
+
+    if cache_refresh_required("redis"):
+        supported_redis_versions = get_redis_supported_versions()
+    else:
+        supported_redis_versions = read_supported_versions_from_cache("redis")
+
+    redis_extensions = []
+    redis_environments_with_failure = []
+
+    for extension in config.get("extensions").keys():
+
+        if config.get("extensions").get(extension).get("type") == "redis":
+            redis_extensions.append(config["extensions"][extension])
+
+    for redis_extension in redis_extensions:
+        environments = redis_extension.get("environments", {})
+        if isinstance(environments, dict):
+            for environment in environments.keys():
+                env_config = environments.get(environment, {})
+                if isinstance(env_config, dict):
+                    redis_engine_version = env_config.get("engine")
+                    if redis_engine_version not in supported_redis_versions:
+                        redis_environments_with_failure.append(
+                            [extension, environment, redis_engine_version]
+                        )
+        else:
+            click.secho(f'Invalid Redis Configuration Block "{extension}"', fg="red")
+
+    if len(redis_environments_with_failure) > 0:
+        for version_failure in redis_environments_with_failure:
+            click.secho(
+                f'Redis version for environment "{version_failure[1]}" is not in the list of supported Redis Versions',
+                fg="red",
+            )
+
+
+def _validate_opensearch_versions(config):
+
+    supported_opensearch_versions = []
+
+    if cache_refresh_required("opensearch"):
+        supported_opensearch_versions = get_opensearch_supported_versions()
+    else:
+        supported_opensearch_versions = read_supported_versions_from_cache("opensearch")
+
+    opensearch_extensions = []
+    opensearch_environments_with_failure = []
+
+    for extension in config.get("extensions", {}).keys():
+
+        if config.get("extensions").get(extension).get("type") == "opensearch":
+            opensearch_extensions.append(config["extensions"][extension])
+
+    for opensearch_extension in opensearch_extensions:
+
+        environments = opensearch_extension.get("environments", {})
+        if isinstance(environments, dict):
+            for environment in environments.keys():
+                env_config = environments.get(environment, {})
+                if isinstance(env_config, dict):
+                    opensearch_engine_version = env_config.get("engine")
+                    if opensearch_engine_version not in supported_opensearch_versions:
```

Comment:
This is 5 levels of nesting. Could be useful to break into methods to make it clear what each level is trying to do, and help turn some of the if statements into guard clauses.

Reply:
It looks very similar to the 5 deep nesting above too. There may be an opportunity to extract some or all of it out to shared method(s).

Reply:
Agreed  @WillGibson @benn-dbt it reads badly - will push some changes